Tests/HashTreeCollectionsTests/TreeDictionary.Keys Tests.swift

```diff
@@ -105,5 +105,45 @@ class TreeDictionaryKeysTests: CollectionTestCase {
       }
     }
   }
+  
+  func test_isEqual_exhaustive() {
+    withEverySubset("a", of: testItems) { a in
+      let x = TreeDictionary<RawCollider, Int>(
+        uniqueKeysWithValues: a.lazy.map { ($0, 2 * $0.identity) })
+      let u = Set(a)
+      expectEqualSets(x.keys, u)
+      withEverySubset("b", of: testItems) { b in
+        let y = TreeDictionary<RawCollider, Int>(
+          uniqueKeysWithValues: b.lazy.map { ($0, -$0.identity - 1) })
+        let v = Set(b)
+        expectEqualSets(y.keys, v)
+
+        let reference = u == v
+        print(reference)
+
+        expectEqual(x.keys == y.keys, reference)
+      }
+    }
+  }
+  
+  func test_Equatable_Hashable() {
+    let samples: [[TreeDictionary<Int, Int>]] = [
+      [[:], [:]],
+      [[1: 100], [1: 100]],
+      [[2: 200], [2: 200]],
+      [[3: 300], [3: 300]],
+      [[100: 1], [100: 1]],
+      [[1: 1], [1: 1]],
+      [[100: 100], [100: 100]],
+      [[1: 100, 2: 200], [2: 200, 1: 100]],
+      [[1: 100, 2: 200, 3: 300],
+       [1: 100, 3: 300, 2: 200],
+       [2: 200, 1: 100, 3: 300],
+       [2: 200, 3: 300, 1: 100],
+       [3: 300, 1: 100, 2: 200],
+       [3: 300, 2: 200, 1: 100]]
+    ]
+    checkHashable(equivalenceClasses: samples.map { $0.map { $0.keys }})
```

Comment:
https://gist.github.com/vanvoorden/5895caeef3417920e262e4c85edcbf0c

@lorentey I'm seeing a ton of test failures here… is there an obvious bug here you think I am missing?

Reply:
```
diff --git a/Tests/HashTreeCollectionsTests/TreeSet Tests.swift b/Tests/HashTreeCollectionsTests/TreeSet Tests.swift
index aa3f675..b309f07 100644
--- a/Tests/HashTreeCollectionsTests/TreeSet Tests.swift	
+++ b/Tests/HashTreeCollectionsTests/TreeSet Tests.swift	
@@ -684,6 +684,9 @@ class TreeSetTests: CollectionTestCase {
       [
         ["a", "b"], ["b", "a"],
       ],
+      [
+        ["a", "b"], ["b", "a"],
+      ],
       [
         ["a", "d"], ["d", "a"],
       ],
```
@lorentey Trying to repeat a pair of sets in this test leads to a failure… is it possible that `checkHashable` helper is expected to fail when we duplicate members?

Reply:
Oh, yes! That is precisely it: the test verifies not only that items in the same equivalence class compare equal, but also that items in other classes don't.

We'll also want the test to verify that `==` and `hash(into:)` ignore the values, so we should exercise those differences too. (I suppose these two cases do that, though!)

Here is one possible way to update things:

```
let samples: [[TreeDictionary<Int, Int>]] = [
       [[:], [:]],
       [[1: 100], [1: 100], [1: 1], [1: 2]],
       [[2: 200], [2: 201]],
       [[3: 300], [3: 301]],
       [[100: 1], [100: 1], [100: 100], [100: 100]],
       [[1: 100, 2: 200], [2: 200, 1: 100], [1: 100, 2: 201]],
       [[1: 100, 2: 200, 3: 300],
        [1: 100, 3: 300, 2: 200],
        [2: 200, 1: 100, 3: 300],
        [2: 200, 3: 300, 1: 100],
        [3: 300, 1: 100, 2: 200],
        [3: 300, 2: 200, 1: 100]]
     ]
```

(I haven't tried this, so beware of thinkos. 😅)